src/xlsx.js

```diff
@@ -0,0 +1,99 @@
+export class ExcelWorkbook {
+  constructor(workbook) {
+    Object.defineProperty(this, "_", {value: workbook});
+  }
+  sheetNames() {
+    return this._.worksheets.map((sheet) => sheet.name);
+  }
+  sheet(name, {range, headers = false} = {}) {
+    const sheet = this._.getWorksheet(
+      typeof name === "number" ? this.sheetNames()[name] : name + ""
+    );
+    if (!sheet) throw new Error(`Sheet not found: ${name}`);
+    return extract(sheet, {range, headers});
+  }
+}
+
+function extract(sheet, {range, headers}) {
+  let [[c0, r0], [c1, r1]] = parseRange(range, sheet);
+  const seen = new Set();
+  const names = [];
+  const headerRow = headers && sheet._rows[r0++];
+  function name(n) {
+    if (!names[n]) {
+      let name = (headerRow ? valueOf(headerRow._cells[n]) : AA(n)) || AA(n);
+      while (seen.has(name)) name += "_";
+      seen.add((names[n] = name));
+    }
+    return names[n];
+  }
+  if (headerRow) for (let c = c0; c <= c1; c++) name(c);
+
+  const output = new Array(r1 - r0 + 1).fill({});
+  for (let r = r0; r <= r1; r++) {
+    const _row = sheet._rows[r];
+    if (!_row || !_row.hasValues) continue;
+    const row = (output[r - r0] = {});
+    for (let c = c0; c <= c1; c++) {
+      const value = valueOf(_row._cells[c]);
+      if (value !== null && value !== undefined) row[name(c)] = value;
+    }
+  }
+
+  output.columns = names.filter(() => true);
```

Comment:
What is this doing?

Reply:
The goal is to add a `columns: ["A", "B", …]` property to the data array, like d3.csv and FileAttachment().csv().

The filtering… maybe a placeholder for removing all the columns that have no value in the dataset? 🤔

Reply:
The filtering is compacting the sparseness. Could be written more clearly as `names.filter((d) => d !== undefined)`, but `filter` by itself skips over the sparse gaps. And yeah, adds the `columns` property so that `Inputs.table` will display the columns in the correct order as row object fields could come in sparsely/out of order.